core-web/libs/sdk/analytics/src/lib/shared/analytics.utils.ts

```diff
@@ -0,0 +1,79 @@
+import { ANALYTICS_SOURCE_TYPE } from './analytics.constants';
+import { DotAnalyticsConfig, PageViewEvent } from './analytics.model';
+
+/**
+ * Retrieves analytics attributes from a given script element.
+ *
+ * @return {DotAnalyticsConfig | null} - The analytics attributes or null if there are no valid attributes present.
+ */
+export const getDataAnalyticsAttributes = (location: Location): DotAnalyticsConfig => {
+    const script = getAnalyticsScriptTag();
+
+    const attributes = {
+        server: script.getAttribute('data-analytics-server') || location.href,
+        debug: script.hasAttribute('data-analytics-debug'),
+        autoPageView: script.hasAttribute('data-analytics-auto-page-view'),
+        key: script.getAttribute('data-analytics-key') || ''
+    };
+
+    return attributes;
+};
+
+/**
+ * Retrieves the analytics script tag from the document.
+ *
+ * @returns {HTMLScriptElement} - The analytics script tag.
+ */
+export const getAnalyticsScriptTag = (): HTMLScriptElement => {
+    const scripts = document.querySelector('script[data-analytics-server]');
+
+    if (!scripts) {
+        throw new Error('Analytics script not found');
+    }
+
+    return scripts as HTMLScriptElement;
+};
+
+/**
+ * Creates the data for a page view event.
+ *
+ * @param {string} event_type - The type of event.
+ * @param {Location} location - The location object.
+ * @returns {PageViewEvent} - The data for the page view event.
+ */
+export function createAnalyticsPageViewData(event_type: string, location: Location): PageViewEvent {
+    const urlParams = new URLSearchParams(location.search);
+    const utmParams = {
+        source: urlParams.get('utm_source') || undefined,
+        medium: urlParams.get('utm_medium') || undefined,
+        campaign: urlParams.get('utm_campaign') || undefined,
+        id: urlParams.get('utm_id') || undefined
+    };
```

Comment:
we can use an enum for this keys and probably make it dynamic

For example: 

```typescript
enum UTM_KEYS {
    source = 'utm_source',
    medium = 'utm_medium',
    campaign = 'utm_campaign',
    id = 'utm_id',
    cheesecake = 'utm_cheesecake' // Random one
}

const urlParams = new URLSearchParams(location.search);

const utmParams = Object.entries(UTM_KEYS).reduce((acc, [key, value]) => {
    acc[key] = urlParams.get(value);
    return acc;
}, {});

```

Reply:
Changed to a utility function, it is easier to test and clearer.